### PR TITLE
feature for get gasPrice for custom json rpc

### LINKED ISSUE #7263: Metamask should give a setting for get gas price directly from node using JSONRPC
**What problem are you trying to solve?**
There is a bunch of blockchains based on ethereum, like Hyperledger Besu, this blockchains  have own gas price and inclusive its possible define 0 as gas price. 
  
**Describe the solution you'd like**
Metamask should be able when adding a new custom network define the eth_gasPrice if the source is from default metamask source or request to node.
In the Advanced configuration if this setting is defined should be permit 0 gas price.

**Additional context**
The JSON RPC is this.
https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_gasprice

If i have some time this week i will implement this if metamask team give-me a green light to this feature. 

Thanks in advance.

### PR DESCRIPTION
**Pull request by helderjnpinto**
Fixes MetaMask/metamask-extension#7263
Maybe need to add calculation for fee based in gasLimit but i can't find the data to do this.

### CODE REVIEW

**Review by danfinlay — CHANGES_REQUESTED** [review]
We would prefer that this parameter be stored per-network, instead of as a global setting across all networks, to allow better multi-network handling.

**Comment by helderjnpinto** [comment]
> We would prefer that this parameter be stored per-network, instead of as a global setting across all networks, to allow better multi-network handling.

yes this is one attribute by network. you can see in the code...

File: ui/app/pages/send/send-content/send-gas-row/send-gas-row.component.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import SendRowWrapper from '../send-row-wrapper'
 import GasFeeDisplay from './gas-fee-display/gas-fee-display.component'
 import GasPriceButtonGroup from '../../../../components/app/gas-customization/gas-price-button-group'
 import AdvancedGasInputs from '../../../../components/app/gas-customization/advanced-gas-inputs'
-
+import { hexToDecimal } from '../../../../helpers/utils/conversions.util'
```
Comment (d10r) [inline ui/app/pages/send/send-content/send-gas-row/send-gas-row.component.js:7]: newline required after this line (linter rule)

**Comment by d10r** [comment]
@helderjnpinto the 3. and 4. CI failure seem to be easy to fix.
One is about a missing newline
```
/home/circleci/project/ui/app/pages/send/send-content/send-gas-row/send-gas-row.component.js
  7:1  error  Expected 1 empty line after import statement not followed by another import  import/newline-after-import
```

and the other one because of tests not having been adapter to the changes.

What causes the first 2 failures (e2e) is less clear to me.

I'd try to fix those myself, but hesitate to do so, because I've no idea how to locally test changes of a browser extension.

**Comment by Gudahtt** [comment]
It's good that this new preference is configured per-network, but I don't think we're tracking the correct thing here. I also don't think we should be calling `eth_gasPrice` in the UI, because the background should already be calling that and initializing each transaction with the result. I don't think there is a need to call it from the UI in any situation.

The only per-network preference we need is whether or not to show the ethgasstation-based gas estimation UI. It should be enabled for mainnet, and disabled otherwise.

**Comment by Gudahtt** [comment]
I'll close this now, since we've already mostly solved this by other means (we now use `eth_gasPrice` exclusively for custom networks).

**Pull request closed without merging**